### PR TITLE
Okhttp tracing support

### PR DESCRIPTION
**Pull request by OlgaMaciaszek**
Fixes gh-36.

### CODE REVIEW

File: spring-cloud-square-okhttp/src/main/java/org/springframework/cloud/square/okhttp/tracing/OkHttpTracingConfiguration.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+/**
+ * Configuration that provides tracing support for OkHttp.
+ *
+ * @author Olga Maciaszek-Sharma
+ */
+@Configuration
+@ConditionalOnClass(TraceContext.class)
+@ConditionalOnBean({ CurrentTraceContext.class, HttpClientHandler.class })
+@ConditionalOnProperty(value = "okhttp.tracing.enabled", havingValue = "true", matchIfMissing = true)
```
Comment (marcingrzejszczak) [inline spring-cloud-square-okhttp/src/main/java/org/springframework/cloud/square/okhttp/tracing/OkHttpTracingConfiguration.java:36]: In general all spring related projects should have `spring.` prefix.

**Review by marcingrzejszczak — APPROVED** [review]

File: spring-cloud-square-okhttp/src/main/java/org/springframework/cloud/square/okhttp/tracing/OkHttpTracingConfiguration.java (hunk 1 above)
Comment (OlgaMaciaszek) [reply spring-cloud-square-okhttp/src/main/java/org/springframework/cloud/square/okhttp/tracing/OkHttpTracingConfiguration.java:36]: That's right; I did it this way, cause there were other existing properties prefixed like this, but I'll convert all of them in a separate PR.

File: docs/src/main/asciidoc/spring-cloud-square.adoc [hunk 2]
```diff
@@ -165,3 +165,9 @@ NOTE: You can create various instances of `WebClient.Builder` with different set
 === Retrofit Reactor support
 
 When `ReactorCallAdapterFactory` is on the classpath (provided by `retrofit2-reactor-adapter` dependency), we also instantiate a bean of this type, by using available `Scheduler` (if present). You can disable this functionality in properties by setting the value of `retrofit.reactor.enabled` to `false`.
+
+=== Spring Cloud Sleuth support
+
+We provide tracing support via Spring Cloud Sleuth integration. If you add `spring-cloud-starter-sleuth` to your project setup, tracing will be provided out of the box, either using the `WebClient` tracing integration provided Spring Cloud Sleuth or via `OkHttpClient` tracing integration provided by the `spring-cloud-square-okhttp` module.
```
Comment (Buzzardo) [inline docs/src/main/asciidoc/spring-cloud-square.adoc:171]: Change to:

We provide tracing support by using Spring Cloud Sleuth integration. If you add `spring-cloud-starter-sleuth` to your project setup, tracing is provided either by the `WebClient` tracing integration (provided Spring Cloud Sleuth) or by the `OkHttpClient` tracing integration (provided by the `spring-cloud-square-okhttp` module).

**Review by Buzzardo — CHANGES_REQUESTED** [review]
Two broad rules that we try to follow: Avoid "via" (part of our initiative to avoid Latin terms) and remove "out of the box" (because it rarely adds any information).

Thanks.

**Review by spencergibb — APPROVED** [review]

**Pull request merged**